### PR TITLE
Expand {Path,OsStr}::{to_str,to_string_lossy} doc examples.

### PR DESCRIPTION
**Pull request by frewsxcv**

### CODE REVIEW

File: src/libstd/ffi/os_str.rs [hunk 1]
```diff
@@ -259,6 +259,30 @@ impl OsStr {
...
+    /// ```
+    ///
+    /// Calling `to_str` on an `OsStr` with invalid unicode:
+    ///
+    /// ```
+    /// use std::ffi::OsStr;
+    ///
+    /// unsafe {
+    ///     let bytes = b"foo\xc3".to_vec();
+    ///     let string = String::from_utf8_unchecked(bytes);
```
Comment (ollie27) [inline src/libstd/ffi/os_str.rs]: Isn't this Undefined Behaviour?

**Comment by ubsan** [comment]
Please no :(

File: src/libstd/ffi/os_str.rs (hunk 1 above)
Comment (frewsxcv) [inline src/libstd/ffi/os_str.rs]: Is it still UB if it doesn't get represented as a string?

**Comment by frewsxcv** [comment]
> Please no :(

In retrospect, I agree. I overlooked that what I am doing is UB.

**Comment by frewsxcv** [comment]
Let me know if the latest changes look better.

**Comment by ubsan** [comment]
Much nicer :)

**Comment by frewsxcv** [comment]
Squashed.

**Comment by GuillaumeGomez** [comment]
Good for me as well. Thanks for the review @ubsan! ;)

@bors: r+ rollup

**Pull request merged**